perfkitbenchmarker/benchmark_spec.py

```diff
@@ -228,6 +233,34 @@ def ConstructVirtualMachines(self):
       self.vm_groups[group_name] = vms
       self.vms.extend(vms)
 
+  def ConstructSparkService(self):
+    if self.config.spark_service is None:
+      return
+
+    providers.LoadProvider(self.config.spark_service.cloud)
+    spark_service_spec = self.config.spark_service
+    service_type = spark_service_spec.spark_service_type
+    if service_type == spark_service.PKB_MANAGED:
```

Comment:
How about changing GetSparkService to GetSparkServiceClass(cloud, service_type) so you don't need this if/else